### PR TITLE
FLINK-17142: Bump ORC dependency version to 1.5.6

### PR DESCRIPTION
**Pull request by zenfenan**
## What is the purpose of the change
Flink's ORC dependency is 1.4.3 which is little outdated so it makes sense to upgrade to a more latest stable version.

## Brief change log

- Update `orc.version` to 1.5.6
- Fixed a failing test case.
- Added `org.apache.hadoop:hadoop-hdfs` to the exclusions in `flink-orc` and `flink-orc-nohive` module since `orc-core:1.5.6` has `hadoop-hdfs` as a compile time dependency

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? (**not applicable**)

### CODE REVIEW

File: flink-formats/flink-orc-nohive/src/main/java/org/apache/flink/orc/nohive/vector/AbstractOrcNoHiveVector.java [hunk 1]
```diff
@@ -138,9 +138,11 @@ private static BytesColumnVector createBytesVector(int batchSize, Object value)
 			bcv.isNull[0] = true;
 			bcv.isRepeating = true;
 		} else {
-			bcv.fill(value instanceof byte[] ?
-					(byte[]) value :
-					value.toString().getBytes(StandardCharsets.UTF_8));
+			byte[] bytes = value instanceof byte[] ?
```
Comment (JingsongLi) [inline flink-formats/flink-orc-nohive/src/main/java/org/apache/flink/orc/nohive/vector/AbstractOrcNoHiveVector.java:141]: Yes, good job, it should same to `AbstractOrcColumnVector`

**Comment by kl0u** [comment]
@JingsongLi if you are ok with this change could you also merge the PR?

**Comment by zentol** [comment]
@JingsongLi Did you check whether any licensing needs to be updated? (i.e., is orc bundled anywhere?)

**Comment by zenfenan** [comment]
@zentol I checked if NOTICE or LICENSE has to be updated but I don't think if that is needed. Appreciate if you or @JingsongLi or @kl0u could check. 

@all If everything checks out, can we close this one? Thanks :)

**Comment by JingsongLi** [comment]
I found `mvn clean install -Phive-2.2.0` in flink-connector-hive can not work.
@zenfenan can you create a `hive.orc.version=1.4.3` in flink-connector-hive. And modify hive-related orc to this one? (Include flink-sql-connecotr-hive-...)

**Comment by zentol** [comment]
It is bundled in the hive 1.2/2.2 sql-connector modules; the corresponding entries in their NOTICE files (src/main/resources/META-INF/NOTICE) may have to be updated,

**Comment by zenfenan** [comment]
@JingsongLi I think it is the case for all hive versions <= 2.2.0. Will introduce a new property with `1.4.3` and provide that as the version for the required profiles.

**Comment by zenfenan** [comment]
@zentol `flink-sql-connector-hive-xyz` modules still use 1.4.3 so I am not touching their NOTICE files.

**Pull request merged**